website/static/js/oauthAddonNodeConfig.js

```diff
@@ -0,0 +1,244 @@
+                    return this.options.onPickFolder.call(this, evt, item);
+                }.bind(this),
+                resolveLazyloadUrl: function(item) {
+                    return item.data.urls.folders;
+                }
+            }
+        );
+    },
+    toggleShare: function() {
+        if (this.currentDisplay() === this.SHARE) {
+            this.currentDisplay(null);
+        } else {
+            // Clear selection
+            this.cancelSelection();
+            this.currentDisplay(this.SHARE);
+            this.activateShare();
+        }
+    },
+    fetchEmailList: function(){
+        var self = this;
+
+        var ret = $.Deferred();
+        var promise = ret.promise();
+        if(!self.loadedEmails()){
+            $.ajax({
+                url: self.urls().emails,
+                type: 'GET',
+                dataType: 'json'
+            }).done(function(res){
+                self.loadedEmails(true);
+                ret.resolve(res.results.emails);
+            }).fail(function(xhr, status, error){
+                Raven.captureMessage('Could not GET ' + self.addonName + ' email list', {
+                    url: self.urls().emails,
+                    textStatus: status,
+                    error: error
+                });
+                ret.reject(xhr, status, error);
+            });
+        }
+        else{
+            ret.resolve(self.emails());
+        }
+        return promise;
+    },
+    activateShare: function() {
+        var self = this;
+        self.fetchEmailList()
+            .done(function(emails){
+                self.emails(emails);
+            });
+        var $copyBtn = $(self.selector).find('.copyBtn');
+        new ZeroClipboard($copyBtn);
+    },
+    _updateCustomFields: function(settings){
+        this.validCredentials(settings.validCredentials);
+    },
+     /**
+     * Allows a user to create an access token from the nodeSettings page
+     */
+    connectAccount: function() {
+        this.options.connectAccount.call(this);
+    },
+    importAuth: function(){
+        var self = this;
+
+        self.updateAccounts(function() {
+            if (self.accounts().length > 1) {
+                bootbox.prompt({
+                    title: 'Choose ' + self.addonName + ' Access Token to Import',
+                    inputType: 'select',
+                    inputOptions: ko.utils.arrayMap(
+                        self.accounts(),
+                        function(item) {
+                            return {
+                                text: item.name,
+                                value: item.id
+                            };
+                        }
+                    ),
+                    value: self.accounts()[0].id,
+                    callback: (self.connectExistingAccount.bind(self))
+                });
+            } else {
+                bootbox.confirm({
+                    title: 'Import ' + self.addonName + ' Access Token?',
```

Comment:
It's not accessible to the user, `self.addName` is defined by the constructor and then never touched.